src/cloudHandler.js

```diff
@@ -37,6 +44,31 @@ var handler = function (params, success, failure) {
 };
 
 /**
+ * Get client id 
+ * 
+ * @param {*} payload object to add parameter 
+ */
+function getClientId(payload) {
+    if (window && window.device) {
+        return window.device.uuid;
+    }
+    if (navigator && navigator.device) {
+        return navigator.device.uuid;
+    }
+    if (window && window.localStorage) {
+        var clientId = window.localStorage.getItem(CLIENT_ID_TAG);
+        if (!clientId) {
+            clientId = uuidGenerator();
+            localStorage.setItem(CLIENT_ID_TAG, clientId);
+        }
+        return clientId;
+    } else {
+        throw Error("Cannot create and store client id");
```

Comment:
Best would be using external plugins but I don't think that should be done here. I will add it to my RN library for sync if you agree.